packages/transport/src/core/types/PasswordProtectionCreationParameters.ts

```
@@ -14,16 +31,33 @@ export class PasswordProtectionCreationParameters extends Serializable implement
     @serialize()
     public password: string;
 
+    @validate({ nullable: true, min: 0, max: 99, customValidator: (v) => (!Number.isInteger(v) ? "This value must be an integer." : undefined) })
+    @serialize({ any: true })
+    public passwordLocationIndicator?: number;
+
     public static from(value: IPasswordProtectionCreationParameters): PasswordProtectionCreationParameters {
         return this.fromAny(value);
     }
 
-    public static create(params: { password: string; passwordIsPin?: true } | undefined): PasswordProtectionCreationParameters | undefined {
+    public static create(
+        params: { password: string; passwordIsPin?: true; passwordLocationIndicator?: PasswordLocationIndicator } | undefined
+    ): PasswordProtectionCreationParameters | undefined {
         if (!params) return;
 
+        const passwordLocationIndicator =
+            params.passwordLocationIndicator !== undefined ? this.mapPasswordLocationIndicatorOptionToNumber(params.passwordLocationIndicator) : undefined;
+
         return PasswordProtectionCreationParameters.from({
             password: params.password,
-            passwordType: params.passwordIsPin ? `pin${params.password.length}` : "pw"
+            passwordType: params.passwordIsPin ? `pin${params.password.length}` : "pw",
+            passwordLocationIndicator: passwordLocationIndicator
         });
     }
+
+    private static mapPasswordLocationIndicatorOptionToNumber(value: PasswordLocationIndicator): number {
+        if (typeof value === "number") return value;
+
+        const numericValue = PasswordLocationIndicatorOptions[value];
+        return numericValue;
```

this should throw when not found.